commit f44f3d0a9e9fbfb800a2a98c02bac5772baedf90
Author: Matthew Wilcox <willy@infradead.org>
Date:   Tue Sep 11 21:27:44 2018 -0700

    filesystem-dax: Fix use of zero page
    
    commit b90ca5cc32f59bb214847c6855959702f00c6801 upstream.
    
    Use my_zero_pfn instead of ZERO_PAGE(), and pass the vaddr to it instead
    of zero so it works on MIPS and s390 who reference the vaddr to select a
    zero page.
    
    Cc: <stable@vger.kernel.org>
    Fixes: 91d25ba8a6b0 ("dax: use common 4k zero page for dax mmap reads")
    Signed-off-by: Matthew Wilcox <willy@infradead.org>
    Reviewed-by: Ross Zwisler <ross.zwisler@linux.intel.com>
    Signed-off-by: Dan Williams <dan.j.williams@intel.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/fs/dax.c b/fs/dax.c
index 641192808bb6..94f9fe002b12 100644
--- a/fs/dax.c
+++ b/fs/dax.c
@@ -1007,21 +1007,12 @@ static vm_fault_t dax_load_hole(struct address_space *mapping, void *entry,
 {
 	struct inode *inode = mapping->host;
 	unsigned long vaddr = vmf->address;
-	vm_fault_t ret = VM_FAULT_NOPAGE;
-	struct page *zero_page;
-	pfn_t pfn;
-
-	zero_page = ZERO_PAGE(0);
-	if (unlikely(!zero_page)) {
-		ret = VM_FAULT_OOM;
-		goto out;
-	}
+	pfn_t pfn = pfn_to_pfn_t(my_zero_pfn(vaddr));
+	vm_fault_t ret;
 
-	pfn = page_to_pfn_t(zero_page);
 	dax_insert_mapping_entry(mapping, vmf, entry, pfn, RADIX_DAX_ZERO_PAGE,
 			false);
 	ret = vmf_insert_mixed(vmf->vma, vaddr, pfn);
-out:
 	trace_dax_load_hole(inode, vmf, ret);
 	return ret;
 }
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1247_f44f3d0a9e9fbfb800a2a98c02bac5772baedf90.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1247_f44f3d0a9e9fbfb800a2a98c02bac5772baedf90.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
